File: Frontend/src/types/idea.ts
export type IdeaStatus =
  | "Prototype"
  | "Incubating"
  | "Exploration"
  | "Pinned"
  | "Research"
  | "Buildable"
  | "Experimental"
  | "Capture"
  | "Archive";

export interface Idea {
  id: string;
  title: string;
  status: IdeaStatus;
  description: string;
  tags: string[];
  progress: number;
  resources: number;
  notes: number;
  updated: string;
  activity: "hot" | "active" | "quiet" | "new";
  code: string;
  importance: number;
  texture: string;
  problem: string;
}

export interface ResourcePreview {
  type: string;
  title: string;
  meta: string;
  description: string;
}

export interface TimelineEntry {
  time: string;
  text: string;
}

export interface GallerySlide {
  title: string;
  caption: string;
  art: string;
}

export type KanbanLaneId = "todo" | "progress" | "completed";

export interface KanbanTask {
  id: string;
  title: string;
  points: number;
}

export type KanbanBoard = Record<KanbanLaneId, KanbanTask[]>;

File: Backend/app/schemas/idea.py
from typing import Literal

from pydantic import BaseModel, Field


IdeaStatus = Literal[
    "Prototype",
    "Incubating",
    "Exploration",
    "Pinned",
    "Research",
    "Buildable",
    "Experimental",
    "Capture",
    "Archive",
]
Activity = Literal["hot", "active", "quiet", "new"]


class IdeaCreate(BaseModel):
    title: str = Field(min_length=1, max_length=180)
    description: str = ""
    problem: str = ""
    tags: list[str] = []
    status: IdeaStatus = "Capture"
    source: str | None = None
    quick_note: str | None = None


class IdeaUpdate(BaseModel):
    title: str | None = None
    status: IdeaStatus | None = None
    description: str | None = None
    problem: str | None = None
    tags: list[str] | None = None
    progress: int | None = None
    activity: Activity | None = None
    importance: int | None = None
    texture: str | None = None


class IdeaOut(BaseModel):
    id: str
    title: str
    status: IdeaStatus
    description: str
    tags: list[str]
    progress: int
    resources: int
    notes: int
    updated: str
    activity: Activity
    code: str
    importance: int
    texture: str
    problem: str
    memoryState: str
<<<<<<< HEAD
=======
    coverUrl: str | None = None
>>>>>>> 6f1c767a5b6ce400673ed3b3987875468dd9fa04

